packages/nextra-theme-blog/src/basic-layout.tsx

```diff
@@ -0,0 +1,27 @@
+import Head from 'next/head'
+import React, { ReactNode, useEffect } from 'react'
+import { useBlogContext } from './blog-context'
+import { HeadingContext } from './mdx-theme'
+import { getTitle } from './utils/title'
+
+export const BasicLayout = ({ children }: { children: ReactNode }) => {
+  const { config, opts } = useBlogContext()
+  const { title, pageTitle } = getTitle({ config, opts })
+  const ref = React.useRef<HTMLHeadingElement>(null)
+  const { hasJsxInH1 } = opts
+  const portalHeading = hasJsxInH1
+  return (
+    <article className="container prose prose-sm dark:prose-dark md:prose">
+      <Head>
+        <title>{title}</title>
+        {config.head?.({ title, meta: opts.meta })}
+      </Head>
+      <HeadingContext.Provider value={ref}>
+        {portalHeading ? <h1 ref={ref} /> : null}
+        {!portalHeading ? <h1>{pageTitle}</h1> : null}
```

Comment:
I mentioned ternary for this part
```suggestion
        {portalHeading ? <h1 ref={ref} /> : <h1>{pageTitle}</h1>}
```

Reply:
also, you don't need to apply suggestions manually just click to Commit Suggestion button
<img width="212" alt="image" src="https://user-images.githubusercontent.com/7361780/181514538-88dabadf-5e62-4f75-bb13-ba72f2aeae7e.png">

Reply:
<img width="1126" alt="Screen Shot 2022-07-28 at 23 57 51" src="https://user-images.githubusercontent.com/15878786/181583769-68ee24c3-d972-4a57-85ed-18b96936b3d1.png">

if you do that, it will cause runtime error here. 

Like i said before. You have to make sure everytime portalHeading changes, a new h1 ReactNode will be created instead of reusing the current one